apps/studio/components/interfaces/Integrations/Vault/Secrets/__tests__/EditSecretModal.test.tsx

```diff
@@ -0,0 +1,130 @@
+import { describe, expect, it, beforeEach, vi } from 'vitest'
+import { mockAnimationsApi } from 'jsdom-testing-mocks'
+import { screen, waitFor, fireEvent } from '@testing-library/dom'
+import userEvent from '@testing-library/user-event'
+import { render } from 'tests/helpers'
+import EditSecretModal from '../EditSecretModal'
+import { addAPIMock } from 'tests/lib/msw'
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from 'ui'
+import { Edit3, MoreVertical } from 'lucide-react'
+import { useState } from 'react'
+import { ProjectContextProvider } from 'components/layouts/ProjectLayout/ProjectContext'
+
+const secret = {
+  id: '47ca58b4-01c5-4a71-8814-c73856b02e0e',
+  name: 'test',
+  description: 'new text',
+  secret: 'NASR0SoksURJ0OorMJ9FzraTzcqSWk5u1PQa2r4c3w9rUVc=',
+  created_at: '2025-07-13 12:57:47.62208+00',
+  updated_at: '2025-07-13 14:51:37.818223+00',
+}
+
+const Page = ({ onClose }: { onClose: () => void }) => {
+  const [modal, setModal] = useState<string | null>(null)
+  const renderModal = () => {
+    switch (modal) {
+      case `edit`:
+        return (
+          <EditSecretModal
+            secret={secret}
+            onClose={() => {
+              setModal(null)
+              onClose()
+            }}
+          />
+        )
+      default:
+        return null
+    }
+  }
+  return (
+    <ProjectContextProvider projectRef="default">
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button title="Manage Secret" type="text" className="px-1" icon={<MoreVertical />} />
+        </DropdownMenuTrigger>
+        <DropdownMenuContent side="bottom" align="end" className="w-32">
+          <DropdownMenuItem className="space-x-2" onClick={() => setModal(`edit`)}>
+            <Edit3 size="14" />
+            <p>Edit</p>
+          </DropdownMenuItem>
+        </DropdownMenuContent>
+      </DropdownMenu>
+
+      {renderModal()}
+    </ProjectContextProvider>
```

Comment:
this much boilerplate may not be necessary, but I wanted to ensure that we're getting the right behavior integrating with `DropdownMenu`, where the two popovers can have some complicated mounting/unmounting behaviors if not done correctly